### PR TITLE
Capan modal

### PR DESCRIPTION
**Pull request by capan**
1. Add geolocation permission modal.

image

### CODE REVIEW

**Comment by orabazu** [comment]
@capan could you please resolve conflict & add an image of what is coming?

File: src/components/Modal/Modal.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import './Modal.scss';
+const Modal = ({ handleClose, show, children }) => {
+    const showHideClassName = show ? "modal modal-display-block" : "modal modal-display-none";
+    return (
+      <div className={showHideClassName}>
+        <section className="modal-main">
+          {children}
+          <button onClick={handleClose}>close</button>
+        </section>
+      </div>
```
Comment (orabazu) [inline src/components/Modal/Modal.js]: We are not given many many options to user here. If user click close button gets an empty page which is kind of intimidating. I guess we need something like `Enable Geeolocation` button to trigger permission API like here https://developer.mozilla.org/en-US/docs/Web/API/Permissions_API/Using_the_Permissions_API

File: src/components/Modal/Modal.scss [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  
+  .modal-main {
+    position:fixed;
+    background: white;
+    width: 80%;
+    height: auto;
+    top:50%;
+    left:50%;
+    transform: translate(-50%,-50%);
+  }
```
Comment (orabazu) [inline src/components/Modal/Modal.scss]: we have sass enabled build, you probably wanna write in a format like 
`.modal{
 &.-main {
  ...
  }
}`

File: src/components/Modal/Modal.scss [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  
+  .modal-display-block {
+    display: block;
+  }
```
Comment (orabazu) [inline src/components/Modal/Modal.scss]: we can move this inside of modal using sass syntax

File: src/components/Modal/Modal.scss [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  
+  .modal-display-none {
```
Comment (orabazu) [inline src/components/Modal/Modal.scss:24]: we can move this inside of modal using sass syntax

File: src/App.js [hunk 5]
```diff
@@ -1,15 +1,61 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './App.css';
 
-import Navbar from '../src/components/Navbar/Navbar'
-import Map from '../src/components/Map/Map'
+import Map from '../src/components/Map/Map';
+import Modal from '../src/components/Modal/Modal';
+import Navbar from '../src/components/Navbar/Navbar';
```
Comment (orabazu) [inline src/App.js]: Useless path segments for "../src/components/Map/Map", should be "./components/Map/Map"

File: src/App.js [hunk 6]
```diff
@@ -1,15 +1,61 @@
...
 
 
 function App() {
+  let [appState, setAppState] = useState({ showModal: true, geolocationPermission: false });
```
Comment (orabazu) [inline src/App.js]: Are we changing these values anywhere? If not you might use const instead

File: src/App.js [hunk 7]
```diff
@@ -1,15 +1,61 @@
...
+
+  const hideModal = () => {
+    setAppState({
+      ...appState,
+      showModal: false
+    });
+  };
   return (
     <div className="App">
       <Navbar></Navbar>
```
Comment (orabazu) [inline src/App.js]: Can be self closing like `<Navbar/>`

**Review by orabazu — CHANGES_REQUESTED** [review]

File: src/components/Modal/Modal.js (hunk 1 above)
Comment (capan) [reply src/components/Modal/Modal.js]: Permission API is automatically triggered when location is tried to be retrieved. In any case user should be able to use the app. Permissions can be asked once and afterwards users can decide whether their positions will be showed on the map.

File: src/components/Modal/Modal.scss (hunk 2 above)
Comment (capan) [reply src/components/Modal/Modal.scss]: not seen much needed

File: src/components/Modal/Modal.scss (hunk 3 above)
Comment (capan) [reply src/components/Modal/Modal.scss]: I agree, fixed !

File: src/App.js (hunk 6 above)
Comment (capan) [reply src/App.js]: Correct, changed to const!

File: src/App.js (hunk 7 above)
Comment (capan) [reply src/App.js]: 👍

**Comment by orabazu** [comment]
LGTM

**Review by orabazu — APPROVED** [review]

**Pull request merged**